docs/companion.md

```diff
@@ -3,3 +3,693 @@ sidebar_position: 4
+
+##### `s3.getKey(req, filename, metadata)`
+
+Get the key name for a file. The key is the file path to which the file will be uploaded in your bucket.
+This option should be a function receiving three arguments:
+
+* `req`, the HTTP request, for _regular_ S3 uploads using the `@uppy/aws-s3` plugin.
+  This parameter is _not_ available for multipart uploads using the `@uppy/aws-s3-multipart` plugin;
+* `filename`, the original name of the uploaded file;
+* `metadata`, user-provided metadata for the file.
+  See the [`@uppy/aws-s3`](https://uppy.io/docs/aws-s3/#metaFields) docs. The `@uppy/aws-s3-multipart` plugin unconditionally sends all metadata fields, so they all are available here.
+
+This function should return a string `key`.
+The `req` parameter can be used to upload to a user-specific folder in your bucket, for example:
+
+```js
+app.use(authenticationMiddleware)
+app.use(uppy.app({
+  providerOptions: {
+    s3: {
+      getKey: (req, filename, metadata) => `${req.user.id}/${filename}`,
+      /* auth options */
+    },
+  },
+}))
+```
+
+The default implementation returns the `filename`,
+so all files will be uploaded to the root of the bucket as their original file name.
+
+```js
+app.use(uppy.app({
+  providerOptions: {
+    s3: {
+      getKey: (req, filename, metadata) => filename,
+    },
+  },
+}))
+```
+
+#### `COMPANION_AWS_SECRET_FILE`
+
+Using a secret file instead of [`s3.secret`](#s3secret-companion_aws_secret).
+A secret file is a file that only contains the secret.
+ 
+#### `COMPANION_AWS_USE_ACCELERATE_ENDPOINT`
+
+Enable S3 [Transfer Acceleration](https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html).
+
+#### `COMPANION_AWS_EXPIRES`
+
+Set `X-Amz-Expires` query parameter in the presigned urls (in seconds, default: 300).
+
+#### `COMPANION_AWS_ACL`
+
+Set a [Canned ACL](https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl) for uploaded objects.
+
+#### `customProviders`
+
+This option enables you to add custom providers along with the already supported providers.
+See [adding custom providers](#how-to-add-custom-providers) for more information.
+
+#### `debug` `COMPANION_DEBUG`
+
+A boolean flag to tell Companion whether to log useful debug information while running.
+
+#### `logClientVersion`
+
+A boolean flag to tell Companion whether to log its version upon startup.
+
+#### `metrics` `COMPANION_HIDE_METRICS`
+ 
+A boolean flag to tell Companion whether to provide an endpoint `/metrics` with Prometheus metrics.
+
+#### `streamingUpload` `COMPANION_STREAMING_UPLOAD`
+
+A boolean flag to tell Companion whether to enable streaming uploads.
+If enabled, it will lead to _faster uploads_ because companion will start uploading at the same time as downloading using `stream.pipe`.
+If `false`, files will be fully downloaded first, then uploaded.
+Defaults to `false`. 
```

Comment:
```suggestion
A boolean flag to tell Companion whether to enable streaming uploads. If enabled, it will lead to _faster uploads_ because companion will start uploading at the same time as downloading using `stream.pipe`. If `false`, files will be fully downloaded first, then uploaded. Defaults to `false`, but we recommended enabling it, especially if you're expecting to upload large files. In future versions the default might change to `true`.
```